.github/workflows/Readme.md

```diff
@@ -0,0 +1,297 @@
+    C --> |Fail|G
+    D --> |Dissatisfied|G  
+    E --> |Disapproved|G[Cleanup Job]
+
+    C3-.->|pull tar|E
+    E -.->H[dockerhub]
+    E -.->I[Quay.io]
+    E -.->J[Acacia S3]      
+
+    subgraph Ella Deploy
+        J-.->|Approved & ARM64|K[(WekaFS)]
+        K-->|Singularity|L[SIF File]
+    end
+
+    subgraph RUNNER storage speed-up
+        C[Build and Push Job] -.-> |push tar|C3[(Local storage)]
+        C3-.->|pull tar|D[Scan and Report Job]
+    end
+    
+    %% Styling for clarity
+    style A fill:#f9f,stroke:#333,stroke-width:2px
+    style F fill:#f66,stroke:#333,stroke-width:2px
+    style G fill:#bbf,stroke:#333,stroke-width:2px
+    style B1 fill:#fff,stroke:#333,stroke-width:2px
+    style B2 fill:#fff,stroke:#333,stroke-width:2px
+    style B3 fill:#fff,stroke:#333,stroke-width:2px
+    style C1 fill:#add8e6,stroke:#333,stroke-width:2px
+    style H fill:#add8e6,stroke:#333,stroke-width:2px
+    style I fill:#add8e6,stroke:#333,stroke-width:2px
+    style J fill:#add8e6,stroke:#333,stroke-width:2px
+```
+
+- [Trigger Logic](#trigger-logic)
+- [Runner Logic](#runner-logic)
+- [Prerequisites](#prerequisites)
+- [Variables and Secrets Configuration](#variables-and-secrets-configuration)
+- [Workflow Overview](#workflow-overview)
+  - [PREPARE-job](#prepare-job)
+  - [BUILD-AND-PUSH-job](#build-and-push-job)
+  - [SCAN-AND-REPORT-job](#scan-and-report-job)
+  - [APPROVE-AND-PUSH-job](#approve-and-deploy-job)
+  - [CLEANUP-job](#cleanup-job)
+  - [Deploy-job](#deploy-job) 
+- [Cache Logic](#cache-logic)
+- [Cross-Platform Logic](#cross-platform-logic)
+- [Conclusion](#conclusion)
+
+## Trigger Logic
+
+The workflow is triggered on every `push` event to the repository. However, it only proceeds under the following conditions:
+
+- **Single Dockerfile Modification**: Exactly **ONLY** one Dockerfile can be modified in the commit.
+- **Specific Label Present**: 
+    - The modified Dockerfile must contain the label `org.opencontainers.image.compilation=auto`.
+    - The modified Docerfile must contain the laber for arch `LABEL org.opencontainers.image.arch=x86` or `LABEL org.opencontainers.image.arch=arm`
+
+If these conditions are not met, the workflow exits gracefully without performing any further actions. This ensures that builds are only triggered for intentional and specific changes.
+
+## Runner logic
+**Which instance will be used for compilation for execution?**
+
+The "Determine Runner Label" step selects the appropriate runner for building the project based on the required architecture and runner availability. 
+- If an **ARM compilation is required** (`LABEL org.opencontainers.image.arch=arm`) and an **ARM-architecure runner is online**, it uses the ARM runner for the build. 
+- Otherwise, it defaults to using the `X64` architecture with `Buildx`(`X64` or `ARM`). 
+- We didn't use the instance guarded methoed, i.e., upto only one `ARM` label runner and only one `X64` can be kept running online simultaneously.
```

Comment:
For a compiled execution, we assume it requires two jobs, A and B, to be completed sequentially. In reality, on GitHub action, each job is stateless, meaning the B job doesn't know the output of the A job. They rely on uploading and downloading artifacts to transfer data. To speed things up, I used the local disk as a cache. However, if multiple runners with the same name are running simultaneously (e.g., for ARM), issues arise. For instance, one runner may be assigned the A job, while another directly starts the B job, breaking the pipeline.

The solution is straightforward: use the runner group method. However, since I currently only have a single laptop deployed on my internal network for compilation, I’m unable to test the group compilation approach.

It also means two separate pushes will be **queued**.